### PR TITLE
Refactor/hydration

### PR DESCRIPTION
**Pull request by cameronRomo**
## What is the change? 
* Refactor `Hydration.js` and `test/Hydration-test.js`

## What does it fix? 
* Both files have better SRP and testing to reflect new methods

## Is this a fix or a feature?  
* This is a refactor of both files

## Where should the reviewer start? 
* Major changes to note are on:
 - `Hydration.js`:
    * line 6
    * line 23
  - `test/Hydration-test.js`:
    * line 15
    * line 137
    * 157

## How should this be tested?
* Pull down locally and run tests

### CODE REVIEW

File: src/Hydration.js [hunk 1]
```diff
@@ -2,21 +2,36 @@ class Hydration {
   constructor(hydrationData) {
     this.hydrationData = hydrationData;
   }
+
+  getAverageOfValues(data) {
```
Comment (mdflynn) [inline src/Hydration.js:6]: Good catch. Nice way to DRY up the class.

File: src/Hydration.js [hunk 2]
```diff
@@ -14,15 +14,24 @@ class Hydration {
     let perDayUserHydration = this.hydrationData.filter((data) => id === data.userID);
     return this.getAverageOfValues(perDayUserHydration);
   }
+
   calculateDailyOunces(id, date) {
     let findOuncesByDate = this.hydrationData.find((data) => id === data.userID && date === data.date);
     return findOuncesByDate.numOunces;
   }
+
+  getOuncesPerDay(loggedData) {
```
Comment (okalimbet) [inline src/Hydration.js:23]: nice catch! definitely looks cleaner with this helper function

**Review by mdflynn — APPROVED** [review]
This class looks to be well refactored.

**Review by okalimbet — APPROVED** [review]

File: src/Hydration.js [hunk 3]
```diff
@@ -2,11 +2,17 @@ class Hydration {
   constructor(hydrationData) {
     this.hydrationData = hydrationData;
   }
+
+  getAverageOfValues(data) {
+    return data.reduce((total, value) => {
```
Comment (okalimbet) [inline src/Hydration.js:7]: Great reuse of the function we created in user-repo. It became handy!

File: test/hydration-test.js [hunk 4]
```diff
@@ -119,59 +147,13 @@ describe('Hydration', function() {
 
   it('should find the water intake for a user on a specified date', function() {
     expect(hydration.calculateDailyOunces(1, "2019/06/15")).to.equal(37);
-    expect(hydration.calculateDailyOunces(4, "2019/04/15")).to.equal(36);
   });
 
   it('should find water intake by day for first week', function() {
-    const user3 = new User({
```
Comment (okalimbet) [inline test/hydration-test.js:126]: Nice cleanup of repetitive variables. The test file looks better now! We need to fix white scapes in userrepo-test file to be consistent

**Review by okalimbet — APPROVED** [review]

**Pull request merged**